Prox/Prox/PlaceCarousel/PlaceCarouselViewController.swift

```diff
@@ -330,43 +332,112 @@ class PlaceCarouselViewController: UIViewController {
+                self.openDetail(forPlace: place)
+            }
+        }
+    }
+
+    func inAppNotificationToastProviderDidDismiss(_ toast: InAppNotificationToastProvider) {
+        self.notificationToastProvider = nil
+    }
 }
 
 extension PlaceCarouselViewController: PlaceDataSource {
 
     func nextPlace(forPlace place: Place) -> Place? {
         // if the place isn't in the list, make the first item in the list the next item
-        guard let currentPlaceIndex = places.index(where: {$0 == place}) else {
-            return places.count > 0 ? places[places.startIndex] : nil
+        let allPlaces = places + Array(placesWithEvents)
+        guard let currentPlaceIndex = allPlaces.index(where: {$0 == place}) else {
+            return allPlaces.count > 0 ? allPlaces[allPlaces.startIndex] : nil
         }
 
-        guard currentPlaceIndex + 1 < places.endIndex else { return nil }
+        guard currentPlaceIndex + 1 < allPlaces.endIndex else { return nil }
 
-        return places[places.index(after: currentPlaceIndex)]
+        return allPlaces[allPlaces.index(after: currentPlaceIndex)]
     }
 
     func previousPlace(forPlace place: Place) -> Place? {
-        guard let currentPlaceIndex = places.index(where: {$0 == place}),
-            currentPlaceIndex > places.startIndex else { return nil }
+        let allPlaces = places + Array(placesWithEvents)
+        guard let currentPlaceIndex = allPlaces.index(where: {$0 == place}),
+            currentPlaceIndex > allPlaces.startIndex else { return nil }
 
-        return places[places.index(before: currentPlaceIndex)]
+        return allPlaces[allPlaces.index(before: currentPlaceIndex)]
     }
 
     func numberOfPlaces() -> Int {
-        return places.count
+        return places.count + placesWithEvents.count
     }
 
     func place(forIndex index: Int) throws -> Place {
-        guard index < places.endIndex,
-            index >= places.startIndex else {
+        let allPlaces = places + Array(placesWithEvents)
+        guard index < allPlaces.endIndex,
+            index >= allPlaces.startIndex else {
             throw PlaceDataSourceError(message: "There is no place at index: \(index)")
         }
 
-        return places[index]
+        return allPlaces[index]
     }
 
     func index(forPlace place: Place) -> Int? {
-        return places.index(of: place)
+        let allPlaces = places + Array(placesWithEvents)
+        return allPlaces.index(of: place)
+    }
+
+    func fetchPlace(placeKey: String, withEvent eventKey: String, callback: @escaping (Place?) -> ()) {
+        if let placeIndex = places.index(where: { $0.id == placeKey }) {
+            let place = places[placeIndex]
+            if place.events.contains(where: { $0.id == eventKey }) {
+                callback(place)
+            }
+            let eventProvider = EventsProvider()
+            eventProvider.event(forKey: eventKey) { event in
+                callback(place)
```

Comment:
The event is ignored here. Did you need it for anything? Seems weird to get the event then ignore it.